test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/NormalizedModelElementsCacheTests.cs

```diff
@@ -157,10 +157,115 @@ public void FindTerms_ReturnsNullIfThereAreNoMatches()
             model.AddTerm("NS.Models", "SomeTerm", EdmPrimitiveTypeKind.String);
             model.AddTerm("NS.Models", "OtherTerm", EdmPrimitiveTypeKind.String);
 
-            var cache = new NormalizedSchemaElementsCache(model);
+            var cache = new NormalizedModelElementsCache(model);
 
             var matches = cache.FindTerms("ns.models.some");
             Assert.Null(matches);
         }
+
+        [Theory]
+        [InlineData("People")]
+        [InlineData("people")]
+        public void FindNavigationSources_ReturnsNavigationSourcesThatMatcheName(string name)
+        {
+            var model = new EdmModel();
+            var person = model.AddEntityType("NS", "People");
+            person.AddKeys(person.AddStructuralProperty("Id", EdmPrimitiveTypeKind.Int32));
+
+            var container = model.AddEntityContainer("NS", "Container");
+            var entitySet1 = container.AddEntitySet("People", person);
+            var entitySet2 = container.AddSingleton("peoPle", person);
+            container.AddEntitySet("Persons", person);
+
+            var cache = new NormalizedModelElementsCache(model);
+
+            var matches = cache.FindNavigationSources(name);
+
+            Assert.Equal(2, matches.Count);

```

Comment:
Why do I think this should be 1 match? EntitySet for person is `Persons` and not people..but you are using people to find the navigation sources? Or there is no need of having that persons in the test

Reply:
It's to matches because the test is searching for "People" (and also "people"), and there are two navigation sources that match these keywords, the "People" entity set and the "peoPle" singleton.

The reason for having "Persons" in the test is to verify that navigation sources that do not match the key are not included in the result.